instrumentation/spring/spring-integration-4.1/javaagent/src/test/groovy/SpringIntegrationAndRabbitTest.groovy

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.api.trace.SpanKind.CLIENT
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER
+import static io.opentelemetry.api.trace.SpanKind.PRODUCER
+import static io.opentelemetry.api.trace.SpanKind.SERVER
+import static io.opentelemetry.instrumentation.test.server.ServerTraceUtils.runUnderServerTrace
+
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+
+class SpringIntegrationAndRabbitTest extends AgentInstrumentationSpecification implements WithRabbitProducerConsumerTrait {
+  def setupSpec() {
+    startRabbit()
+  }
+
+  def cleanupSpec() {
+    stopRabbit()
+  }
+
+  def "should cooperate with existing RabbitMQ instrumentation"() {
+    when:
+    // simulate the workflow being triggered by HTTP request
+    runUnderServerTrace("HTTP GET") {
+      producerContext.getBean("producer", Runnable).run()
+    }
+
+    then:
+    assertTraces(2) {
+      trace(0, 7) {
+        span(0) {
+          name "HTTP GET"
+          kind SERVER
+          attributes {}
+        }
+        span(1) {
+          name "producer"
+          childOf span(0)
+          attributes {}
+        }
+        span(2) {
+          name "exchange.declare"
```

Comment:
can you comment in the test which spans are from spring integration instrumentation and which are from rabbitmq instrumentation?

Reply:
Good point, will do.